Test failure with cowsay installed/present (again)

### Summary

This was previously reported under https://github.com/ansible/ansible/issues/83327 but it is still occurring, after that was supposedly fixed.

When running the build/tests for "test_broken_cowsay" the test fails if the cowsay package is present/available on the system:
```=================================== FAILURES ===================================
_____________________ test_display_with_fake_cowsay_binary _____________________
[gw11] linux -- Python 3.12.4 /usr/bin/python

capsys = <_pytest.capture.CaptureFixture object at 0x72031b2442f0>
mocker = <pytest_mock.plugin.MockerFixture object at 0x72031b2445c0>

    def test_display_with_fake_cowsay_binary(capsys, mocker):

        mocker.patch("ansible.constants.ANSIBLE_COW_PATH", "./cowsay.sh")
        mock_popen = MagicMock()
        mock_popen.return_value.returncode = 1
        mocker.patch("subprocess.Popen", mock_popen)
        display = Display()

>       assert not hasattr(display, "cows_available")
E       AssertionError: assert not True
E        +  where True = hasattr(<ansible.utils.display.Display object at 0x720321fa63f0>, 'cows_available')

test/units/utils/display/test_broken_cowsay.py:22: AssertionError
```


### Issue Type

Bug Report

### Component Name

test/units/utils/display/test_broken_cowsay.py

### Ansible Version

```console
$ ansible --version
ansible [core 2.16.8]
  config file = /home/ssilverman/.ansible.cfg
  configured module search path = ['/home/ssilverman/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.12/site-packages/ansible
  ansible collection location = /home/ssilverman/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522] (/usr/bin/python)
  jinja version = 3.1.4
  libyaml = True
```


### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
ANSIBLE_NOCOWS(/home/ssilverman/.ansible.cfg) = True
ANSIBLE_PIPELINING(/home/ssilverman/.ansible.cfg) = True
CONFIG_FILE() = /home/ssilverman/.ansible.cfg
DEFAULT_FORKS(/home/ssilverman/.ansible.cfg) = 30
DEFAULT_HOST_LIST(/home/ssilverman/.ansible.cfg) = ['/home/ssilverman/ansible/inventory']
DEFAULT_LOG_PATH(/home/ssilverman/.ansible.cfg) = /home/ssilverman/ansible/ansible.log
DISPLAY_ARGS_TO_STDOUT(/home/ssilverman/.ansible.cfg) = True
EDITOR(env: EDITOR) = vim
INTERPRETER_PYTHON(/home/ssilverman/.ansible.cfg) = auto_silent
RETRY_FILES_ENABLED(/home/ssilverman/.ansible.cfg) = True
RETRY_FILES_SAVE_PATH(/home/ssilverman/.ansible.cfg) = /home/ssilverman/ansible/retry

CONNECTION:
==========

local:
_____
pipelining(/home/ssilverman/.ansible.cfg) = True

psrp:
____
pipelining(/home/ssilverman/.ansible.cfg) = True

ssh:
___
pipelining(/home/ssilverman/.ansible.cfg) = True

winrm:
_____
pipelining(/home/ssilverman/.ansible.cfg) = True
```


### OS / Environment

Arch Linux

### Steps to Reproduce

Issue with tests/build, not executing playbooks


### Expected Results

I expected the tests to pass, whether or not cowsay was present.

### Actual Results

```console
build errors/failures if cowsay is present/installed
```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

--- a/test/units/utils/conftest.py
+++ b/test/units/utils/conftest.py
@@ -1,0 +1,18 @@
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import annotations
+
+import pytest
+
+from ansible.utils.display import Display
+
+
+@pytest.fixture()
+def display_resource(request):
+    Display._Singleton__instance = None
+
+    def teardown():
+        Display._Singleton__instance = None
+
+    request.addfinalizer(teardown)

--- a/test/units/utils/display/test_broken_cowsay.py
+++ b/test/units/utils/display/test_broken_cowsay.py
@@ -9,7 +9,7 @@ from ansible.utils.display import Display
 from unittest.mock import MagicMock
 
 
-def test_display_with_fake_cowsay_binary(capsys, mocker):
+def test_display_with_fake_cowsay_binary(capsys, mocker, display_resource):
 
     mocker.patch("ansible.constants.ANSIBLE_COW_PATH", "./cowsay.sh")
     mock_popen = MagicMock()

--- a/test/units/utils/display/test_display.py
+++ b/test/units/utils/display/test_display.py
@@ -8,7 +8,7 @@ from __future__ import annotations
 from ansible.utils.display import Display
 
 
-def test_display_basic_message(capsys, mocker):
+def test_display_basic_message(capsys, mocker, display_resource):
     # Disable logging
     mocker.patch('ansible.utils.display.logger', return_value=None)
 

--- a/test/units/utils/test_display.py
+++ b/test/units/utils/test_display.py
@@ -54,7 +54,7 @@ def test_get_text_width_no_locale(problematic_wcswidth_chars):
     pytest.raises(EnvironmentError, get_text_width, problematic_wcswidth_chars[0])
 
 
-def test_Display_banner_get_text_width(monkeypatch):
+def test_Display_banner_get_text_width(monkeypatch, display_resource):
     locale.setlocale(locale.LC_ALL, '')
     display = Display()
     display_mock = MagicMock()
@@ -67,7 +67,7 @@ def test_Display_banner_get_text_width(monkeypatch):
     assert msg.endswith(stars)
 
 
-def test_Display_banner_get_text_width_fallback(monkeypatch):
+def test_Display_banner_get_text_width_fallback(monkeypatch, display_resource):
     locale.setlocale(locale.LC_ALL, 'C.UTF-8')
     display = Display()
     display_mock = MagicMock()
@@ -80,12 +80,12 @@ def test_Display_banner_get_text_width_fallback(monkeypatch):
     assert msg.endswith(stars)
 
 
-def test_Display_set_queue_parent():
+def test_Display_set_queue_parent(display_resource):
     display = Display()
     pytest.raises(RuntimeError, display.set_queue, 'foo')
 
 
-def test_Display_set_queue_fork():
+def test_Display_set_queue_fork(display_resource):
     def test():
         display = Display()
         display.set_queue('foo')
@@ -96,7 +96,7 @@ def test_Display_set_queue_fork():
     assert p.exitcode == 0
 
 
-def test_Display_display_fork():
+def test_Display_display_fork(display_resource):
     def test():
         queue = MagicMock()
         display = Display()
@@ -110,7 +110,7 @@ def test_Display_display_fork():
     assert p.exitcode == 0
 
 
-def test_Display_display_warn_fork():
+def test_Display_display_warn_fork(display_resource):
     def test():
         queue = MagicMock()
         display = Display()
@@ -124,7 +124,7 @@ def test_Display_display_warn_fork():
     assert p.exitcode == 0
 
 
-def test_Display_display_lock(monkeypatch):
+def test_Display_display_lock(monkeypatch, display_resource):
     lock = MagicMock()
     display = Display()
     monkeypatch.setattr(display, '_lock', lock)
@@ -132,7 +132,7 @@ def test_Display_display_lock(monkeypatch):
     lock.__enter__.assert_called_once_with()
 
 
-def test_Display_display_lock_fork(monkeypatch):
+def test_Display_display_lock_fork(monkeypatch, display_resource):
     lock = MagicMock()
     display = Display()
     monkeypatch.setattr(display, '_lock', lock)

